aten/src/ATen/core/ivalue.cpp

```diff
@@ -156,6 +156,11 @@ void IValue::visit(const std::function<bool (const IValue &)>& visitor) const {
       }
       break;
     }
+    case Tag::PyObject: {
+      IValue contained_value = toPyObjectHolder()->toIValue();
```

Comment:
what if this PyObject does not contains IValue, this will let this function crash?

Reply:
Indeed, it will just propagate the error raised by the other method. For `getSubValues` (the function just below) I don't think this is a problem: until now that function was raising for _all_ PyObjects, now it only raises for some of them. That's an improvement.

However I agree that for `visit` such a behavior could be considered a regression, since before PyObjects were silently skipped over. If you want I can leave the current behavior for `visit`. (I only need `getSubValues` for my use-case).

Reply:
sounds good, let's leave the current behavior for `visit` to make sure no regression.

Reply:
I removed that regression from `visit`. Now, `visit` will still try to convert a PyObject to an IValue, and if that succeeds it will recurse into that IValue. However, if it fails, it silently skips.

I think that this might be the best of both worlds: it adds an improvement to `visit` and keeps it aligned with `getSubValues`, but it also "preserves the spirit" of `visit` of avoiding hard failures.